zebra-consensus/src/transaction.rs

```diff
@@ -425,6 +435,58 @@ where
     ZS: Service<zs::Request, Response = zs::Response, Error = BoxError> + Send + Clone + 'static,
     ZS::Future: Send + 'static,
 {
+    /// Get the UTXOs that are being spent by the given transaction.
+    ///
+    /// `known_utxos` are additional UTXOs known at the time of validation (i.e.
+    /// from previous transactions in the block).
+    ///
+    /// Returns a tuple with a OutPoint -> Utxo map, and a vector of Outputs
+    /// in the same order as the matching inputs in the transaction.
+    async fn spent_utxos(
+        tx: Arc<Transaction>,
+        known_utxos: Arc<HashMap<transparent::OutPoint, OrderedUtxo>>,
+        state: Timeout<ZS>,
+    ) -> Result<
+        (
+            HashMap<transparent::OutPoint, transparent::Utxo>,
+            Vec<transparent::Output>,
+        ),
+        TransactionError,
+    > {
+        let inputs = tx.inputs();
+        let mut spent_utxos = HashMap::new();
+        let mut spent_outputs = Vec::new();
+        for input in inputs {
+            match input {
+                transparent::Input::PrevOut {
+                    outpoint,
+                    unlock_script: _,
+                    sequence: _,
+                } => {
```

Comment:
Optional: this could remove one level of indentation by using an `if let`:

```suggestion
            if let transparent::Input::PrevOut { outpoint, .. } = input {
```

Reply:
Done in b36577315 / #3446